blockchain/state_manager/src/lib.rs

```diff
@@ -184,9 +184,14 @@ where
         round: ChainEpoch,
         entropy: &[u8],
     ) -> anyhow::Result<[u8; 32]> {
-        let chain_rand = ChainRand::new(blocks.to_owned(), self.cs.clone(), self.beacon.clone());
+        let chain_rand = ChainRand::new(
+            self.chain_config.clone(),
+            blocks.to_owned(),
+            self.cs.clone(),
+            self.beacon.clone(),
+        );
         match self.get_network_version(round) {
-            NetworkVersion::V15 | NetworkVersion::V14 => {
+            NetworkVersion::V16 | NetworkVersion::V15 | NetworkVersion::V14 => {
```

Comment:
@LesnyRumcajs, Note we can't use the wildcard pattern here because of `#[non_exhaustive]` on `NetworkVersion` type.

Reply:
I believe this attribute has a separate meaning - you cannot `not` use the wildcard pattern. In this match, it's even used at the end. Or am I missing something?

Reply:
It means we have to use it actually. Even if we have exhaustive pattern matching.

Reply:
If you remove the wildcard case you will have this error:
```
        match self.get_network_version(round) {
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ pattern `_` not covered
```
I think the purpose of `#[non_exhaustive]` is to avoid consumer code to break if they add a new network version variant in the future.

Reply:
Yes, but my point was to use this wildcard to handle everything that is below 14 with `get_beacon_randomness_v3` and not panic. This way we wouldn't have to revisit this match pattern on every upgrade.

Reply:
You mean above 14, no? 🤔

Reply:
ya